fixbackend/workspaces/repository.py

```diff
@@ -420,6 +430,34 @@ async def ack_move_to_free(self, workspace_id: WorkspaceId, user_id: UserId) ->
 
             return evolve(workspace, move_to_free_acknowledged_at=now)
 
+    async def list_overdue_free_tier_cleanup(self, been_in_free_tier_for: timedelta) -> Sequence[Workspace]:
+        """List all workspaces where the free tier cleanup is overdue."""
+        async with self.session_maker() as session:
+            statement = (
+                select(orm.Organization)
+                .where(orm.Organization.tier == ProductTier.Free.value)
+                .where(orm.Organization.free_tier_cleanup_done_at.is_(None))
+                .where(orm.Organization.created_at < utc() - been_in_free_tier_for)
+            )
+            results = await session.execute(statement)
+            workspaces = results.unique().scalars().all()
+            return [ws.to_model() for ws in workspaces]
+
+    async def ack_overdue_free_tier_cleanup(self, workspace_id: WorkspaceId) -> None:
```

Comment:
consider using an approach that does not require maintaining state in the database if possible